ui/desktop/app/routes/scopes/scope/projects/targets/target.js

```diff
@@ -23,6 +23,56 @@ export default class ScopesScopeProjectsTargetsTargetRoute extends Route {
 
   // =methods
 
+  /**
+   * Given an array of hostSources returns an array of host sets
+   * @param {array} hostSources
+   * @returns {[HostSetModel]} hostSets
+   */
+  async getTargetHostSets(hostSources) {
+    let hostSets = [];
+
+    for (const hostSource of hostSources) {
```

Comment:
I'm not sure I understand why you called these with a `for...of` loop which is not concurrent but also we stop at the first error anyways. Is this to try and return any hosts which might have succeeded?

Why not just try catch the `Promise.all` which fails fast since if there's any forbidden it's unlikely any would succeed?

Reply:
@ZedLi 
Ok, I will implement `Promise.all` in both methods.